TSPL.docc/LanguageGuide/ControlFlow.md

```diff
@@ -1440,6 +1440,78 @@ and `distance` is an integer in both patterns ---
 which means that the code in the body of the `case`
 can always access a value for `distance`.
 
+## Patterns
+
+In the previous examples, each switch case includes a pattern
+that indicates what values match that case.
+You can also use a pattern as the condition for an `if` statement.
+Here's what that looks like:
+
+```swift
+let somePoint = (12, 100)
+if case (let x, 100) = somePoint {
+    print("Found a point on the y=100 line, at \(x)")
+}
+// Prints "Found a point on the y=100 line, at 12"
+```
+
+In this code,
+the condition for the `if` statement starts with `case`,
+indicating that the condition is a pattern instead of a Boolean value.
+If the pattern matches,
+the condition for the `if` is considered to be true,
+and the code in the body of the `if` statement runs.
+The patterns you can write after `if case`
+are the same as the patterns you can write in a switch case.
+
+You can also use a pattern in a `for`-`in` loop,
+to give names to parts of the value using a value binding:
+
+```swift
+let points = [(10, 0), (30, 100), (-20, 0)]
+
+for case let (x, y) in points {
+    if y == 0 {
+        print("Found a point on the x-axis at \(x)")
+    }
+}
+// Prints "Found a point on the x-axis at 10"
+// Prints "Found a point on the x-axis at -20"
+```
+
+The `for`-`in` loop above iterates over an array of tuples,
+binding the first and second elements of the tuples
+to the `x` and `y` constants.
+The statements inside the loop can use those constants,
+like the `if` statement that checks whether the point lies on the x-axis.
+
+A `for`-`case`-`in` loop can also include a `where` clause,
+to check for an additional condition.
+The statements inside the loop run
+only when `where` clause matches the current element.
+For example,
+the `for`-`case`-`in` loop below is the same as the `for`-`in` loop above.
+
+```swift
+for case let (x, y) in points where y == 0  {
+    print("Found a point on the x-axis at \(x)")
+}
+// Prints "Found a point on the x-axis at 10"
+// Prints "Found a point on the x-axis at -20"
+```
```

Comment:
This example doesn't need the `case`, either. What if you make this part about "any `for`-`in` loop" having a `where` clause, and capture points along a line that goes through the origin?

```swift
for case let (x, y) in points where x == y || x == -y  {
    print("Found a point (\(x), \(y)) along line through the origin")
}
```